adoc/chapters/programming_interface.adoc

```diff
@@ -1454,1390 +1454,1969 @@ info::context::atomic_fence_scope_capabilities
 
-The SYCL [code]#device# class encapsulates a single SYCL device on which
-<<kernel,kernels>> can be executed.
+The [code]#device# class represents a single SYCL device on which <<kernel,
+kernels>> can be executed.
 
 All member functions of the [code]#device# class are synchronous and errors are
 handled by throwing synchronous SYCL exceptions.
 
 The execution environment for a SYCL application has a fixed number of
 <<root-device, root devices>> which does not vary as the application executes.
 The application can get a list of all these devices via
-[code]#device::get_devices()#, and the order of the device objects is the same
-each time the application calls that function (assuming the parameter to that
+[api]#device::get_devices#, and the order of the device objects is the same each
+time the application calls that function (assuming the parameter to that
 function is the same for each call).
 The [code]#device# class also provides constructors, but constructing a new
 [code]#device# instance merely creates a new object that is a copy of one of the
-objects returned by [code]#device::get_devices()#.
+objects returned by [api]#device::get_devices#.
 
-A SYCL [code]#device# can be partitioned into multiple SYCL devices, by calling
-the [code]#create_sub_devices()# member function template.
-The resulting SYCL [code]#devices# are considered sub devices, and it is valid
+A device can be partitioned into multiple devices, by calling the
+[code]#device::create_sub_devices# member function template.
+The resulting [code]#device# objects are considered sub devices, and it is valid
 to partition these sub devices further.
 The range of support for this feature is <<backend>> and device specific and can
-be queried for through [code]#get_info()#.
+be queried for through [api]#device::get_info#.
 
-The SYCL [code]#device# class provides the common reference semantics (see
-<<sec:reference-semantics>>).
-
-==== Device interface
-
-A synopsis of the SYCL [code]#device# class is provided below.
-The constructors, member functions and static member functions of the SYCL
-[code]#device# class are listed in <<table.constructors.device>>,
-<<table.members.device>> and <<table.staticmembers.device>> respectively.
-The additional common special member functions and common member functions are
-listed in <<sec:reference-semantics>> in
-<<table.specialmembers.common.reference>> and
-<<table.hiddenfriends.common.reference>>, respectively.
+The [code]#device# class provides the common reference semantics as defined in
+<<sec:reference-semantics>>.
 
-// Interface of the device class
-[source,,linenums]
+[source,role=synopsis]
 ----
 include::{header_dir}/device.h[lines=4..-1]
 ----
 
+[[sec:device-ctors]]
+==== Constructors
 
-[[table.constructors.device]]
-.Constructors of the SYCL [code]#device# class
-[width="100%",options="header",separator="@",cols="65%,35%"]
-|====
-@ Constructor @ Description
-a@
-[source]
+.[apititle]#Default constructor#
```

Comment:
Just curious: is this leading `.` on purpose?

Reply:
Yes.  The `.` Asciidoc syntax adds a title to the `[source]` block below.

Reply:
Thanks for the explanation. I have not read the AsciiDoctor documentation for quite a while...